src/System.Text.Json/src/System/Text/Json/Writer/UnicodeScalar.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers.Text
+{
+    /// <summary>
+    /// Represents a 24-bit Unicode scalar value.
+    /// A scalar value is any value in the range [U+0000..U+D7FF] or [U+E000..U+10FFFF].
+    /// </summary>
+    internal struct UnicodeScalar : IComparable<UnicodeScalar>, IEquatable<UnicodeScalar>
+    {
+        /// <summary>
+        /// The Unicode Replacement Character U+FFFD.
+        /// </summary>
+        public static readonly UnicodeScalar s_replacementChar = new UnicodeScalar(0xFFFD);
+
+        /// <summary>
+        /// The integer value of this scalar.
+        /// </summary>
+        public readonly int Value; // = U+0000 if using default init
+
+        /// <summary>
+        /// Constructs a Unicode scalar from the given value.
+        /// The value must represent a valid scalar.
+        /// </summary>
+        public UnicodeScalar(int value)
+            : this((uint)value)
+        {
+            // None of the APIs on this type are guaranteed to produce correct results
+            // if we don't validate the input during construction.
+
+            if (!IsValidScalar((uint)Value))
+            {
+                throw new ArgumentOutOfRangeException(
```

Comment:
The other types have their exception throws in a static Throw method. Add here too?